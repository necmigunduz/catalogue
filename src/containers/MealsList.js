import { useEffect } from 'react';
import PropTypes from 'prop-types';
import getData from '../assets/logic/FetchData';
import Meal from '../components/Meal';
import SearchField from '../components/Search';
import '../assets/styles/css-lib/borders.css';
import '../assets/styles/css-lib/colors.css';
import '../assets/styles/css-lib/positioning.css';
import '../assets/styles/css-lib/fonts.css';

const MealsList = (props) => {
  const {
    meals, filters, fetchMeals, changeFilter, clearFilter,
  } = props;

  const setMeals = async (ingredient, cuisine, from, to) => {
    const mealsArray = await getData(ingredient, cuisine, from, to);
    fetchMeals(mealsArray);
  };

  const handleFilter = (filter, value) => {
    changeFilter(filter, value);
  };

  const handleSubmit = () => {
    setMeals(filters, 0, 99);
  };

  const handleClear = () => {
    clearFilter();
  };

  useEffect(() => {
    setMeals(filters, 0, 99);
  }, []);

  let listOfMeals;
  if (meals.length > 0) {
    listOfMeals = meals.map((meal) => (
      <Meal
        id={meal.id}
        title={meal.title}
        image={meal.image}
        diet={meal.diet}
        health={meal.health}
        cuisine={meal.cuisine}
        dish={meal.dishType}
        meal={meal.mealType}
        key={meal.id}
      />
    ));
  } else {
    listOfMeals = <p className="m-30">No recipe found!</p>;
  }

  return (
    <>
      <div>
        <SearchField
          filters={filters}
          filterHandler={handleFilter}
          submitHandler={handleSubmit}
          clear={handleClear}
        />
        <div className="listOfMeals">
          <h3 className="fs-20 fw-400 bord-1 pad-20 bg-55 color-white">Search Results</h3>
          {listOfMeals}
        </div>
      </div>
    </>
  );
};

MealsList.propTypes = {
  fetchMeals: PropTypes.func.isRequired,
  changeFilter: PropTypes.func.isRequired,
  clearFilter: PropTypes.func.isRequired,
  meals: PropTypes.arrayOf(PropTypes.object).isRequired,
  filters: PropTypes.shape({
    q: PropTypes.string,
    cuisineType: PropTypes.string,
  }),
};

MealsList.defaultProps = {
  filters: PropTypes.shape({
    q: '',
    cuisineType: 'indian',
  }),
};

export default MealsList;
