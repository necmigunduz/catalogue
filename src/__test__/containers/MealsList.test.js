import React from 'react';
import renderer from 'react-test-renderer';
import MealsList from '../../containers/MealsList';

describe('MealsList', () => {
    it('renders properly', () => {
        const tree = renderer
            .create(
                <MealsList
                    meals={[]}
                    filters={{
                        q: '', dietLabels: '', healthLabels:'', cuisineType: '', dishType: '', mealType: '',
                    }}
                    changeFilter={() => {}}
                    fetchMeals={() => {}}
                    clearFilter={() => {}}
                />
            ).toJSON();
            expect(tree).toMatchSnapshot();
    })
})