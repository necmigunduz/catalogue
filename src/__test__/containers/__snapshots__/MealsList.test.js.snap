// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MealsList renders properly 1`] = `
<div>
  <div
    className="m-30"
  >
    <h2
      className="m-10 fw-900 fs-30"
    >
      Search Your Recipe
    </h2>
    <div
      onChange={[Function]}
    >
      <input
        className="size-20"
        id="q"
        placeholder="Enter ingredients and cooking equipment to search for a recipe including them"
        type="text"
        value=""
      />
      <div
        className="m-20 fw-600"
      >
        <hr />
        <h3>
          Choose your filters below
        </h3>
        <div
          className="pad-10"
        >
          <select
            id="dietLabels"
            value=""
          >
            <option>
              Diet
            </option>
            <option>
              Balanced
            </option>
            <option>
              High-fiber
            </option>
            <option>
              High-protein
            </option>
            <option>
              Low-carb
            </option>
            <option>
              Low-fat
            </option>
            <option>
              Low-sodium
            </option>
          </select>
          <div />
        </div>
        <div
          className="pad-10"
        >
          <select
            id="healthLabels"
            value=""
          >
            <option>
              Health
            </option>
            <option>
              Alcohol-free
            </option>
            <option>
              Celery-free
            </option>
            <option>
              Crustacean-free
            </option>
            <option>
              Dairy-free
            </option>
            <option>
              Egg-free
            </option>
            <option>
              Fish-free
            </option>
            <option>
              Fodmap-free
            </option>
            <option>
              Gluten-free
            </option>
            <option>
              Immuno-supportive
            </option>
            <option>
              Keto-friendly
            </option>
            <option>
              Kidney-friendly
            </option>
            <option>
              Kosher
            </option>
            <option>
              Low-fat-tabs
            </option>
            <option>
              Low-potasium
            </option>
            <option>
              Low-sugar
            </option>
            <option>
              Lupine-free
            </option>
            <option>
              Mediterranean
            </option>
            <option>
              Mustard-free
            </option>
            <option>
              No-oil-added
            </option>
            <option>
              Paleo
            </option>
            <option>
              Peanut-free
            </option>
            <option>
              Pescatarian
            </option>
            <option>
              Pork-free
            </option>
            <option>
              Red-meat-free
            </option>
            <option>
              Sesame-free
            </option>
            <option>
              Shellfish-free
            </option>
            <option>
              Soy-free
            </option>
            <option>
              Sugar-conscious
            </option>
            <option>
              Tree-nut-free
            </option>
            <option>
              Vegan
            </option>
            <option>
              Vegetarian
            </option>
            <option>
              Wheat-free
            </option>
          </select>
          <div />
        </div>
        <div
          className="pad-10"
        >
          <select
            id="cuisineType"
            value=""
          >
            <option>
              Cuisine
            </option>
            <option>
              American
            </option>
            <option>
              Asian
            </option>
            <option>
              British
            </option>
            <option>
              Caribbean
            </option>
            <option>
              Central Europe
            </option>
            <option>
              Chinese
            </option>
            <option>
              Eastern Europe
            </option>
            <option>
              French
            </option>
            <option>
              Indian
            </option>
            <option>
              Italian
            </option>
            <option>
              Japanese
            </option>
            <option>
              Kosher
            </option>
            <option>
              Mediterranean
            </option>
            <option>
              Mexican
            </option>
            <option>
              Middle Eastern
            </option>
            <option>
              Nordic
            </option>
            <option>
              South American
            </option>
            <option>
              South East Asian
            </option>
          </select>
          <div />
        </div>
        <div
          className="pad-10"
        >
          <select
            id="dishType"
            value=""
          >
            <option>
              Dish
            </option>
            <option>
              alcohol
            </option>
            <option>
              biscuits and cookies
            </option>
            <option>
              bread
            </option>
            <option>
              cereals
            </option>
            <option>
              condiments and sauces
            </option>
            <option>
              drinks
            </option>
            <option>
              desserts
            </option>
            <option>
              egg
            </option>
            <option>
              main course
            </option>
            <option>
              omelet
            </option>
            <option>
              pancake
            </option>
            <option>
              preps
            </option>
            <option>
              preserve
            </option>
            <option>
              salad
            </option>
            <option>
              sandwiches
            </option>
            <option>
              soup
            </option>
            <option>
              starter
            </option>
          </select>
          <div />
        </div>
        <div
          className="pad-10"
        >
          <select
            id="mealType"
            value=""
          >
            <option>
              Meal
            </option>
            <option>
              Breakfast
            </option>
            <option>
              Lunch
            </option>
            <option>
              Dinner
            </option>
            <option>
              Snack
            </option>
            <option>
              Teatime
            </option>
          </select>
          <div />
        </div>
        <hr />
      </div>
      <a
        className="m-30"
        href="#search"
        onClick={[Function]}
        type="button"
      >
        Click to Search
      </a>
      <button
        id="clear"
        onClick={[Function]}
        type="button"
      >
        Clear All
      </button>
    </div>
  </div>
  <div
    className="listOfMeals"
  >
    <h3>
      Search Results
    </h3>
    <p
      className="m-30"
    >
      No recipe found!
    </p>
  </div>
</div>
`;
